gulpfile.js

```diff
@@ -68,46 +60,62 @@ function lint() {
     .pipe(gjslint.reporter('console'))
 }
 
-function bundle(browserify, env) {
-  return browserify
-    .transform(babelify)
-    .transform(envify, env)
-    .bundle()
-    .on('error', gutil.log.bind(gutil, 'Browserify Error'))
-    .pipe(source(jsFileName))
-    .pipe(buffer())
-    .pipe(sourcemaps.init({loadMaps: true}))
-      .pipe(uglify({
-        preserveComments: 'some'
-      }))
-      .on('error', gutil.log)
-    .pipe(sourcemaps.write('./'))
-    .pipe(gulp.dest('dist'));
+function bundle(format) {
+  return esperanto.bundle({
+    base: '.',
+    entry: entryFileName,
+  }).then(function(bundle) {
+    return bundle[format]({
+      strict: true,
+      sourceMap: 'inline',
+      name: 'IncrementalDOM'
+    });
+  });
 }
 
-function js() {
-  var b_plain = browserify(customBrowserifyOpts);
-
-  return bundle(b_plain, {});
+function js(done) {
+  bundle('toUmd').then(function(gen) {
+    file(artifactName + '.js', gen.code, {src: true})
+      .pipe(sourcemaps.init({loadMaps: true}))
+      .pipe(babel())
+      .pipe(sourcemaps.write('./'))
+      .pipe(gulp.dest('dist'))
+      .on('error', done)
+      .on('end', done);
+  }).catch(done);
 }
 
 function jsWatch() {
-  var opts = assign({}, watchify.args, customBrowserifyOpts);
-  var b_watch = watchify(browserify(opts));
+  gulp.watch(srcs, ['js']);
+}
 
-  b_watch.on('update', bundle.bind(null, b_watch));
-  b_watch.on('log', gutil.log);
+function jsDist(done) {
+  process.env.NODE_ENV = 'production';
 
-  return bundle(b_watch, {});
+  bundle('toUmd').then(function(gen) {
+    file(artifactName + '.js', gen.code, {src: true})
+      .pipe(sourcemaps.init({loadMaps: true}))
+      .pipe(babel())
+      .pipe(uglify({
+        preserveComments: 'some'
+      }))
+      .pipe(sourcemaps.write('./'))
+      .pipe(gulp.dest('dist'))
+      .on('error', done)
+      .on('end', done);
+  }).catch(done);
 }
 
-function jsDist() {
-  var b_plain = browserify(customBrowserifyOpts);
+function jsClosure(done) {
+  bundle('toCjs').then(function(gen) {
+    var code = 'goog.module(\'incrementaldom\');' + gen.code;
```

Comment:
This'll break the `'use strict'` statement. How about a single regex replace for the first newline?